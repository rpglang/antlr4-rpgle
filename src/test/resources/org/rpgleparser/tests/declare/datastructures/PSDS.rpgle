04      dcl-ds PgmDs psds qualified ;
05         PgmName *proc ;
06         Status *status ;
07      end-ds ;