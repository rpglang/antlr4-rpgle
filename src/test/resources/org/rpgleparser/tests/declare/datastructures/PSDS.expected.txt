/*===TOKENS===*/
HIDDEN:LEAD_WS5_Comments <04   >
DS_DataStructureStart    <dcl-ds>
ID                       <PgmDs>
KEYWORD_PSDS             <psds>
KEYWORD_QUALIFIED        <qualified>
FREE_SEMI                <;>
HIDDEN:LEAD_WS5_Comments <05   >
KEYWORD_PGMNAME          <PgmName>
ID                       <*proc>
FREE_SEMI                <;>
HIDDEN:LEAD_WS5_Comments <06   >
ID                       <Status>
ID                       <*status>
FREE_SEMI                <;>
HIDDEN:LEAD_WS5_Comments <07   >
DS_DataStructureEnd      <end-ds>
FREE_SEMI                <;>
/*===TREE===*/
(r
  (statement
    (dcl_ds
      dcl-ds
      (identifier (free_identifier (idOrKeyword PgmDs)))
      (keyword (keyword_psds psds))
      (keyword (keyword_qualified qualified))
      ;
      (dcl_ds_field
        (identifier (free_identifier (idOrKeyword PgmName)))
        (identifier (free_identifier (idOrKeyword *proc)))
        ;
      )
      (dcl_ds_field
        (identifier (free_identifier (idOrKeyword Status)))
        (identifier (free_identifier (idOrKeyword *status)))
        ;
      )
      (end_dcl_ds end-ds)
      ;
    )
  )
)
/*======*/